sdk/python/test_data/pipelines/pipeline_with_nested_conditions_yaml.yaml

```diff
@@ -302,6 +302,7 @@ deploymentSpec:
         - '{{$.inputs.parameters[''msg'']}}'
         image: python:alpine3.6
 pipelineInfo:
+  description: Shows how to use dsl.Condition().
   name: conditional-execution-pipeline
```

Comment:
Can we at least have one pipeline that sets both `name` and `display_name` to make sure we are not mixing the two?

Reply:
Yes. Added a unit test and a compilation test for this case.